### ISSUE TITLE
black formatter not working when file contains several comments

### ISSUE THREAD

**Issue by Kenza98** [psf/black#4419]
error message:
Oh no! 💥 💔 💥
1 file failed to reformat.

--------------------------------------------------------------

```
for i in range(N):
    for j in range(M):
        """conditions = [
        timestamps[i] is not None,
        not np.isnan(depth[j]),
        not np.isnan(lats[i]),
        not np.isnan(longs[i]),
        not np.isnan(temperature[i, j])
]
if all(conditions):
"""
        list_temp_measures.append(
            {
                # assert
                "timestamp": timestamps[i].isoformat(),
                "depth": floatdepth[j],
                "latitude": lats[i],
                "longitude": longs[i],
                "temperature": temperature[i, j],
            }
        )
```

---------------------------------------------------------------

This works though:
-------------------------------------------------------------
```

"""conditions = [
        timestamps[i] is not None,
        not np.isnan(depth[j]),
        not np.isnan(lats[i]),
        not np.isnan(longs[i]),
        not np.isnan(temperature[i, j])]
if all(conditions):
"""

for i in range(N):
    for j in range(M):

        list_temp_measures.append({
                # assert
                "timestamp": timestamps[i].isoformat(),
                "depth": floatdepth[j],
                "latitude": lats[i],
                "longitude": longs[i],
                "temperature": temperature[i, j],}
        )

```
-----------------------------------------------------------------------
---output msg:
All done! ✨ 🍰 ✨
1 file reformatted.

and gives : 
-------------------------------------------------------------------
```

for i in range(N):
    for j in range(M):

        list_temp_measures.append(
            {
                # assert
                "timestamp": timestamps[i].isoformat(),
                "depth": floatdepth[j],
                "latitude": lats[i],
                "longitude": longs[i],
                "temperature": temperature[i, j],
            }
        )

```
--------------------------------------------------------------------

**Comment by JelleZijlstra** [comment]
That's not true, I misread the code. However, I can't reproduce a crash on Black 24.4.2 and Python 3.12.

**Comment by Kenza98** [comment]
Python 3.11.6
[image: Philippe's face]
KENZA BENJELLOUN
Étudiante en Doctorat au département Applied Data Science and Artificial
Intelligence à l'Université de Trieste, Italie
+33 6 22 16 64 62 <+33782395967>


Le mar. 30 juil. 2024 à 15:00, Jelle Zijlstra ***@***.***> a
écrit :

**Comment by JelleZijlstra** [comment]
This looks like the issue fixed by #4290 in version 24.4.0, are you on the latest release?
